pkg/deployment/reconcile/action_resign_leadership.go

```diff
@@ -0,0 +1,150 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
+//
+
+package reconcile
+
+import (
+	"context"
+
+	"github.com/arangodb/go-driver"
+	"github.com/arangodb/kube-arangodb/pkg/util/arangod"
+
+	"github.com/arangodb/kube-arangodb/pkg/util/errors"
+
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1"
+	"github.com/rs/zerolog"
+)
+
+func init() {
+	registerAction(api.ActionTypeResignLeadership, newResignLeadershipAction)
+}
+
+func newResignLeadershipAction(log zerolog.Logger, action api.Action, actionCtx ActionContext) Action {
+	a := &actionResignLeadership{}
+
+	a.actionImpl = newActionImplDefRef(log, action, actionCtx, shutdownMemberTimeout)
+
+	return a
+}
+
+type actionResignLeadership struct {
+	actionImpl
+}
+
+func (a *actionResignLeadership) Start(ctx context.Context) (bool, error) {
+	log := a.log
+	group := a.action.Group
+	m, ok := a.actionCtx.GetMemberStatusByID(a.action.MemberID)
+	if !ok {
+		log.Error().Msg("No such member")
+		return true, nil
+	}
+
+	if a.actionCtx.GetSpec().Mode.Get() != api.DeploymentModeCluster {
+		log.Debug().Msg("Resign only allowed in cluster mode")
+		return true, nil
+	}
+
+	client, err := a.actionCtx.GetDatabaseClient(ctx)
+	if err != nil {
+		log.Error().Err(err).Msgf("Unable to get client")
+		return true, errors.WithStack(err)
+	}
+
+	switch group {
+	case api.ServerGroupDBServers:
+		cluster, err := client.Cluster(ctx)
+		if err != nil {
+			log.Error().Err(err).Msgf("Unable to get cluster client")
+			return true, errors.WithStack(err)
+		}
+
+		var jobID string
+		jobCtx := driver.WithJobIDResponse(ctx, &jobID)
+		log.Debug().Msg("Temporary shutdown, resign leadership")
+		if err := cluster.ResignServer(jobCtx, m.ID); err != nil {
+			log.Debug().Err(err).Msg("Failed to resign server")
+			return true, errors.WithStack(err)
+		}
+
+		m.CleanoutJobID = jobID
+
+		if err := a.actionCtx.UpdateMember(m); err != nil {
+			return true, errors.WithStack(err)
+		}
+
+		return false, nil
+	default:
+		return true, nil
+	}
+}
+
+func (a *actionResignLeadership) CheckProgress(ctx context.Context) (bool, bool, error) {
+	log := a.log
+
+	m, ok := a.actionCtx.GetMemberStatusByID(a.action.MemberID)
+	if !ok {
+		log.Error().Msg("No such member")
+		return true, false, nil
+	}
+
+	agency, err := a.actionCtx.GetAgency(ctx)
+	if err != nil {
+		log.Debug().Err(err).Msg("Failed to create agency client")
+		return false, false, errors.WithStack(err)
+	}
+
+	c, err := a.actionCtx.GetDatabaseClient(ctx)
+	if err != nil {
+		log.Debug().Err(err).Msg("Failed to create member client")
+		return false, false, errors.WithStack(err)
+	}
+
+	jobStatus, err := arangod.CleanoutServerJobStatus(ctx, m.CleanoutJobID, c, agency)
```

Comment:
Is this correct?
We're not doing cleanout

Reply:
It is valid option - Cleanout and Reasign jobs are handled in same way, just trigerred in different way.

Function could be called just FetchJobStatus.